src/app/body/all-products/filters/filters.component.html

```diff
@@ -1,43 +1,79 @@
-      <app-category-filter></app-category-filter>
-      <app-category-filter></app-category-filter>
-      <app-category-filter></app-category-filter>
-    </ul>
-  </div>
+  <form id="form" novalidate [formGroup]="form" (change)="filter()">
+    <div class="filter">
+      <div class="filter_title">Categories</div>
+      <ul class="filter_items">
+        <li *ngFor="let category of categories">
+          <label class="filter_category">
+            <input type="radio" value="{{category}}" name="categories" formControlName="categories" />
+            <span>{{category.charAt(0).toUpperCase() + category.slice(1)}}</span>
+            <span>{{getCategoryAmount(category)}}</span>
+          </label>
+        </li>
+      </ul>
+    </div>
 
-  <div class="filter">
-    <div class="filter_title">Brands</div>
-    <ul class="filter_items">
-      <app-brand-filter></app-brand-filter>
-      <app-brand-filter></app-brand-filter>
-      <app-brand-filter></app-brand-filter>
-      <app-brand-filter></app-brand-filter>
-      <app-brand-filter></app-brand-filter>
-    </ul>
-  </div>
+    <div class="filter">
+      <div class="filter_title">Brands</div>
+      <ul class="filter_items">
+        <li *ngFor="let brand of brands">
+          <label class="filter_brand">
+            <input type="checkbox" value="{{brand}}" (change)="onBrandsCheckboxChange($event)"/>
+            {{brand.charAt(0).toUpperCase() + brand.slice(1)}}
+            <span></span>
+          </label>
+        </li>
+      </ul>
+    </div>
 
-  <div class="filter">
-    <div class="filter_title">Rating</div>
-    <ul class="filter_items">
-      <app-rating-filter></app-rating-filter>
-    </ul>
-  </div>
+    <div class="filter">
+      <div class="filter_title">Rating</div>
+      <ul class="filter_rating">
+        <li *ngFor="let r of rating; let index = index">
+          <label class="filter_brand">
+            <img *ngFor="let star of rating; let i = index" src="assets/{{index + 1 < i + 2 ? 'star-full' : 'star'}}.svg" alt="star">
+            <input type="checkbox" value="{{5 - index}}" (change)="onRatingCheckboxChange($event)"/>
+            <span></span>
+          </label>
+        </li>
+      </ul>
+    </div>
 
-  <div class="filter">
-    <div class="filter_title">Price</div>
-    <app-price-filter></app-price-filter>
-  </div>
+    <div class="filter" formGroupName="price">
+      <div class="filter_title">Price</div>
+      <div class="filter-price">
+        <div class="filter-price_range">
+          <ngx-slider
+            [(value)]="form.controls.price.value.minPrice"
+            [(highValue)]="form.controls.price.value.maxPrice"
+            [options]="sliderOptions"
+          ></ngx-slider>
+          <button (click)="filter()">OK</button>
+        </div>
+        <div class="filter-price_inputs">
+          <label>Min
+            <input type="number" placeholder="0" formControlName="minPrice" [value]="form.controls.price.value.minPrice">
```

Comment:
`[value]="form.controls.price.value.minPrice"` is unneeded.

Reply:
but without it values in inputs not changes
![image](https://user-images.githubusercontent.com/64072558/101360626-90343800-38a6-11eb-8f9f-d536d655ffc0.png)